src/memray/_memray/hooks.h

```diff
@@ -70,6 +94,7 @@ struct SymbolHook
 
     void ensureValidOriginalSymbol()
     {
+#if defined(__linux__)
```

Comment:
So on Linux, we assume that our original function pointer can be wrong, and we attempt to correct it by resolving the symbol ourselves. Remind me why we need that on Linux, and why we don't on Mac?

Reply:
>  Remind me why we need that on Linux, and why we don't on Mac?

Because this only happens on the weird Debian systems where they build with some odd options that cause the GNU linker to give us the PLT address for just "&malloc". This doesn't happen in Mac because dyld doesn't not get confused by this (I checked).

Reply:
Mm. For what it's worth, we may also need `ensureValidOriginalSymbol` (or something like it) if we ever want to patch a symbol that isn't available from our DSO dependencies. We've suggested patching things like `__tls_get_addr` or `PyInit_foomodule` in the past, and for those things, we can't just take `&` and use that as the original address, we need to look it up dynamically.

Something to keep in mind, at least, though we're currently not doing that anywhere.

Reply:
Unfortunately adding support for that is quite a lot of extra work, because of how macOS does it symbol exposure, so I would prefer to not do it for now.